app/controllers/autocompletes_controller.rb

```diff
@@ -0,0 +1,11 @@
+class AutocompletesController < ApplicationController
+  def show
+    render json: DiscoveryEngine::Autocomplete::Complete.new(query).completion_result
+  end
+
+private
+
+  def query
+    params.permit(:q)[:q]
```

Comment:
Is this of any different effect to doing:

```suggestion
    params[:q]
```